File: problems/easy/majority_element.rb
# @param {Integer[]} nums
# @return {Integer}
def majority_element(nums)
  return nil if nums.length == 0 || nums.nil?
  i = 0
  candidate = 0
  count = 0
  while(i<nums.length)
    candidate = nums[i] if count == 0

    if(candidate == nums[i])
      count += 1
    else
      count -= 1
    end
    i += 1
  end
  return candidate
end

File: problems/easy/majority_element.go
// https://leetcode.com/problems/majority-element/submissions/
func majorityElement(nums []int) int {
	return nil if nums == nil || len(nums) == 0
	var candidate int
	var count int
	for i := 0; i < len(nums); i++ {
		if count == 0 {
			candidate = nums[i]
		}
		if nums[i] == candidate {
			count++
		} else {
			count--
		}
	}
	return candidate
}